WindowsError when running tests under Windows

I get the following error when running unit tests in Windows:

`python setup.py tests`

```
============================= test session starts =============================
platform win32 -- Python 2.7.11, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: D:\Tomer\Desktop\temp2\flask-security, inifile: setup.cfg
plugins: cov-2.2.1, pep8-1.0.6, flakes-1.0.1
collected 168 items

setup.py ..
flask_security\__init__.py ..
flask_security\changeable.py ..
flask_security\confirmable.py ..
flask_security\core.py ..
flask_security\datastore.py ..
flask_security\decorators.py ..
flask_security\forms.py ..
flask_security\passwordless.py ..
flask_security\recoverable.py ..
flask_security\registerable.py ..
flask_security\script.py ..
flask_security\signals.py ..
flask_security\utils.py ..
flask_security\views.py ..
tests\conftest.py ..
tests\test_changeable.py ...E
--------------- coverage: platform win32, python 2.7.11-final-0 ---------------
Name                                                         Stmts   Miss  Cover   Missing
------------------------------------------------------------------------------------------
flask_security\__init__.py                                       8      0   100%
flask_security\changeable.py                                    16      0   100%
flask_security\confirmable.py                                   30     15    50%   29-30, 40-47, 55-56, 74, 82-87
flask_security\core.py                                         157     31    80%   195-202, 232-233, 250-251, 299, 302, 325, 336, 346-347, 353, 357, 360, 363, 366, 369, 372, 375, 378, 381, 384, 387, 450
flask_security\datastore.py                                    194    102    47%   20, 23, 26, 38, 43-44, 47, 52-53, 56, 72, 74, 81-83, 89, 93, 97, 110, 118-124, 128-129, 136-139, 146-149, 175, 188, 200, 214-215, 218-228, 231-240, 245, 264-266, 269-279, 282-285, 288-291, 295-301, 309-316, 324-333
flask_security\decorators.py                                   121     74    39%   40-42, 46-57, 61-77, 81-91, 103-110, 128-133, 158-169, 191-198, 220-226, 235
flask_security\forms.py                                        162     53    67%   78-80, 84-86, 137-139, 146-151, 160-162, 165-170, 179-184, 193, 196-201, 220, 223-224, 227-228, 233-234, 236-237, 239-240, 242-243, 245-246, 258-260
flask_security\passwordless.py                                  16      6    63%   32-38, 46, 57
flask_security\recoverable.py                                   33     20    39%   32-39, 47-48, 57-59, 70-78, 87-90
flask_security\registerable.py                                  20     11    45%   27-43
flask_security\script.py                                        60     60     0%   2-132
flask_security\signals.py                                       10      0   100%
flask_security\utils.py                                        190     78    59%   59, 65-79, 89-93, 102-111, 120-123, 135, 138-139, 150-151, 196, 224, 235, 246, 292-293, 311-313, 337-338, 353-374, 389-399, 409-419, 429-439
flask_security\views.py                                        190    119    37%   41-42, 49-50, 71, 82-85, 93-96, 104-136, 144-159, 168-183, 189-204, 212-237, 245-260, 269-290, 335-338, 346, 351-354, 364-367
------------------------------------------------------------------------------------------
TOTAL                                                         1207    569    53%

=================================== ERRORS ====================================
_________________ ERROR at teardown of test_recoverable_flag __________________

>   request.addfinalizer(lambda: os.remove(path))
E   WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\tomer\\appdata\\local\\temp\\pytest-of-Tomer\\pytest-4\\test_recoverable_flag0\\flask-security-test-dbqffakh.db'

tests\conftest.py:188: WindowsError
---------------------------- Captured stderr setup ----------------------------
d:\tomer\desktop\temp2\flask-security\.eggs\flask_sqlalchemy-2.1-py2.7.egg\flask_sqlalchemy\__init__.py:800: UserWarning: SQLALCHEMY_TRACK_MODIFICATIONS adds significant overhead and will be disabled by default in the future.  Set it to True to suppress this warning.
  warnings.warn('SQLALCHEMY_TRACK_MODIFICATIONS adds significant overhead and will be disabled by default in the future.  Set it to True to suppress this warning.')
!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!
===================== 35 passed, 1 error in 3.75 seconds ======================

Process finished with exit code 2

```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -274,7 +274,6 @@ def sqlalchemy_session_datastore(request, app, tmpdir):
         Base.metadata.create_all(bind=engine)
 
     def tear_down():
-        db_session.commit()
         db_session.close()
         os.close(f)
         os.remove(path)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -212,7 +212,10 @@ def sqlalchemy_datastore(request, app, tmpdir):
     with app.app_context():
         db.create_all()
 
-    request.addfinalizer(lambda: os.remove(path))
+    def tear_down():
+        os.close(f)
+        os.remove(path)
+    request.addfinalizer(tear_down)
 
     return SQLAlchemyUserDatastore(db, User, Role)
 
@@ -270,7 +273,12 @@ def sqlalchemy_session_datastore(request, app, tmpdir):
     with app.app_context():
         Base.metadata.create_all(bind=engine)
 
-    request.addfinalizer(lambda: os.remove(path))
+    def tear_down():
+        db_session.commit()
+        db_session.close()
+        os.close(f)
+        os.remove(path)
+    request.addfinalizer(tear_down)
 
     return SQLAlchemySessionUserDatastore(db_session, User, Role)
 
@@ -319,7 +327,12 @@ def peewee_datastore(request, app, tmpdir):
         for Model in (Role, User, UserRoles):
             Model.create_table()
 
-    request.addfinalizer(lambda: os.remove(path))
+    def tear_down():
+        db.close_db(None)
+        os.close(f)
+        os.remove(path)
+
+    request.addfinalizer(tear_down)
 
     return PeeweeUserDatastore(db, User, Role, UserRoles)
 

